pkg/api/pullrequestpayloadqualification/v1/types.go

```diff
@@ -0,0 +1,93 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PullRequestPayloadQualificationRun represents the intent to run a battery of OCP release
+// payload validating jobs
+type PullRequestPayloadQualificationRun struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   PullRequestPayloadTestSpec   `json:"spec"`
+	Status PullRequestPayloadTestStatus `json:"status"`
+}
+
+// PullRequestPayloadTestSpec specifies for which PR the payload qualification run was requested
+// and the list of individual jobs that should be executed
+type PullRequestPayloadTestSpec struct {
+	PullRequest PullRequestUnderTest        `json:"pullRequest"`
```

Comment:
doc please.  mutable, immutable, required?

Reply:
immutable, required

pretty much everything should be immutable I guess - the requestor should just create the CR and then it makes no sense to change it (similar to Prowjobs)